engine/api/src/main/java/io/deephaven/engine/table/hierarchical/RollupTable.java

```diff
@@ -0,0 +1,158 @@
+     *
+     * @return The rollup aggregations
+     */
+    Collection<? extends Aggregation> getAggregations();
+
+    /**
+     * Are the source table's constituent rows included at the lowest level of the rollup?
+     *
+     * @return Whether constituents are included
+     */
+    boolean includesConstituents();
+
+    /**
+     * Node types for a rollup.
+     */
+    enum NodeType {
+
+        /**
+         * Nodes at one of the aggregated (rolled up) levels in the RollupTable. These nodes have column names and types
+         * that result from the {@link #getAggregations() aggregations}.
+         */
+        Aggregated,
+
+        /**
+         * Nodes at the leaf level when {@link #includesConstituents()} is set. These nodes have column names and types
+         * from the {@link #getSource() source} table.
+         */
+        Constituent
+    }
+
+    /**
+     * Get the {@link NodeType} at this RollupTable's leaf level.
+     *
+     * @return The leaf node type
+     */
+    @FinalDefault
+    default NodeType getLeafNodeType() {
+        return includesConstituents() ? NodeType.Constituent : NodeType.Aggregated;
+    }
+
+    /**
+     * Get the group-by columns for the {@link Table#rollup rollup} operation.
+     *
+     * @return The group-by columns
+     */
+    Collection<? extends ColumnName> getGroupByColumns();
+
+    /**
+     * Get the {@link TableDefinition} that should be exposed to node table consumers, e.g. UI-driven snapshots. This
+     * excludes "internal" columns used to organize the rollup or support operations, as well as the
+     * {@link #getRowDepthColumn() row-depth column} and {@link #getRowExpandedColumn() row-expanded column}, but
+     * includes formatting columns.
+     *
+     * @param nodeType The {@link NodeType node type} to get the {@link TableDefinition} for
+     * @return The externally-visible node {@link TableDefinition} for the requested {@code nodeType}
+     * @throws IllegalArgumentException If {@link NodeType#Constituent Constituent} is specified when
+     *         {@code includesConstituents() == false}
+     */
+    TableDefinition getNodeDefinition(@NotNull NodeType nodeType);
+
+    /**
+     * @inheritDoc
+     * @implNote RollupTable implementations of this method have three distinct sets of columns: the "extra" columns,
+     *           the aggregated node columns, and the constituent node columns (if and only if constituents are
+     *           included), included in the listed order. Constituent node columns will have their names mangled to
+     *           disambiguate them from aggregated node columns, since {@link TableDefinition table definitions} do not
+     *           permit name collisions.
+     * @return The externally-visible {@link TableDefinition} for snapshots
+     */
+    TableDefinition getSnapshotDefinition();
+
+    /**
+     * Get the {@link Pair input/output column name pairs} from input (source) column names to output (rollup
```

Comment:
I have updated the `@implNote` tags to try to clarify:

On `getSnapshotDefinition()`:
```
     * @implNote RollupTable implementations of this method have three distinct sets of columns: the "extra" columns,
     *           the aggregated node columns, and the constituent node columns (if and only if constituents are
     *           included), included in the listed order. Constituent node columns will have their names adjusted to
     *           disambiguate them from aggregated node columns, since {@link TableDefinition table definitions} do not
     *           permit name collisions. For example, an implementation might prefix all constituent column names with
     *           {@code "__CONSTITUENT_"}.
```

On `getColumnPairs()`:
```
     * @implNote The input column names will be adjusted to match the disambiguated constituent column names found in
     *           {@link #getSnapshotDefinition() snapshot definitions} for constituent columns. For example, an
     *           implementation might prefix all input column names with {@code "__CONSTITUENT_"}.
```